src/pip/_internal/commands/debug.py

```diff
@@ -43,6 +47,97 @@ def show_sys_implementation():
+        '_vendor',
+        'vendor.txt'
+    )
+
+    with open(vendor_txt_path) as f:
+        # Purge non version specifying lines.
+        # Also, remove any space prefix or suffixes (including comments).
+        lines = [line.strip().split(' ', 1)[0]
+                 for line in f.readlines() if '==' in line]
+
+    # Transform into "module" -> version dict.
+    return dict(line.split('==', 1) for line in lines)  # type: ignore
+
+
+def get_module_from_module_name(module_name):
+    # type: (str) -> ModuleType
+
+    # Module name can be uppercase in vendor.txt for some reason...
+    module_name = module_name.lower()
+    # PATCH: setuptools is actually only pkg_resources.
+    if module_name == 'setuptools':
+        module_name = 'pkg_resources'
+
+    __import__(
+        'pip._vendor.{}'.format(module_name),
+        globals(),
+        locals(),
+        level=0
+    )
+    return getattr(pip._vendor, module_name)
+
+
+def get_vendor_version_from_module(module_name):
+    # type: (str) -> str
+
+    module = get_module_from_module_name(module_name)
+    version = getattr(module, '__version__', None)
+
+    # Some vendored modules (like requests) have a __version__.py file.
+    # If this is the case, we should grab __version__
+    # from the __version__.py file
+    if isinstance(version, ModuleType):
+        version = getattr(version, '__version__', None)
+    if not version:
+        # Try to find version in debundled module info
+        pkg_set = pkg_resources.WorkingSet(
+            [os.path.dirname(getattr(module, '__file__'))]
+        )
+        package = pkg_set.find(pkg_resources.Requirement.parse(module_name))
+        version = getattr(package, 'version', None)
+
+    return version
+
+
+def show_actual_vendor_versions(vendor_txt_versions):
+    # type: (Dict[str, str]) -> None
+    # Logs the actual version and print extra info
+    # if there is a conflict or if the actual version could not be imported.
+
+    for module_name, expected_version in vendor_txt_versions.items():
+        extra_message = ''
+        actual_version = get_vendor_version_from_module(module_name)
+        if not actual_version:
+            extra_message = ' (Unable to locate actual module version, using'\
+                            ' vendor.txt specified version)'
+            actual_version = expected_version
+        elif actual_version != expected_version:
+            extra_message = ' (CONFLICT: vendor.txt suggests version should'\
+                            ' be {})'.format(expected_version)
+
+        logger.info(
+            '{name}=={actual}{extra}'.format(
+                name=module_name,
+                actual=actual_version,
+                extra=extra_message
+            )
+        )
+
+
+def show_vendor_versions():
+    # type: () -> None
+    logger.info('vendored library versions:')
+
+    vendor_txt_versions = create_vendor_txt_map()
+    show_actual_vendor_versions(vendor_txt_versions)
```

Comment:
A `with indent_log():` would be prettier IMHO:
```
vendored library versions:
  appdirs==1.4.3
  CacheControl==0.12.6
 ...
```
vs
```
vendored library versions:
appdirs==1.4.3
CacheControl==0.12.6
```

Reply:
I see, updated the code to include it- nice idea!